### PR TITLE
[AL-4149] Validate metric names against reserved names

### PR DESCRIPTION
**Pull request by dubininsergey**
Validate metric names against the list of reserved ones

### CODE REVIEW

File: labelbox/data/metrics/iou/iou.py [hunk 1]
```diff
@@ -62,7 +62,7 @@ def feature_miou_metric(ground_truths: List[Union[ObjectAnnotation,
         if value is None:
             continue
         metrics.append(
-            ScalarMetric(metric_name="iou", feature_name=key, value=value))
+            ScalarMetric(metric_name="custom_iou", feature_name=key, value=value))
```
Comment (dubininsergey) [inline labelbox/data/metrics/iou/iou.py:65]: as we agreed in Slack discussion, we rename these to `custom_iou`

File: labelbox/data/annotation_types/metrics/scalar.py [hunk 2]
```diff
@@ -28,6 +40,14 @@ class ScalarMetric(BaseMetric):
     value: Union[ScalarMetricValue, ScalarMetricConfidenceValue]
     aggregation: ScalarMetricAggregation = ScalarMetricAggregation.ARITHMETIC_MEAN
 
+    @validator('metric_name')
+    def validate_metric_name(cls, name: str):
+        clean_name = name.lower().strip()
+        if name.lower().strip() in RESERVED_METRIC_NAMES:
+            raise ValueError(f"`{clean_name}` is a reserved metric name. "
+                             "Please provide another value for `metric_name`.")
+        return clean_name
```
Comment (kozikkamil) [inline labelbox/data/annotation_types/metrics/scalar.py:49]: I think we shouldn't force all lowercase, it's just for validation purpose

Comment (dubininsergey) [reply labelbox/data/annotation_types/metrics/scalar.py:42]: I think we should be consistent, if we're validating against lower-cased value then we should store it that way as well

Comment (dubininsergey) [reply labelbox/data/annotation_types/metrics/scalar.py:42]: but on the other hand it will break existing metrics, yeah 🤔

Comment (dubininsergey) [reply labelbox/data/annotation_types/metrics/scalar.py:42]: I'd say validating against lower-cased metric name feels weird, we should just trim it

Comment (dubininsergey) [reply labelbox/data/annotation_types/metrics/scalar.py:42]: @msokoloff1 @kozikkamil wdyt?

Comment (dubininsergey) [reply labelbox/data/annotation_types/metrics/scalar.py]: alright, I think that's not so important

now we return the source name as it's currently in prod version 👌

File: labelbox/data/annotation_types/metrics/scalar.py [hunk 3]
```diff
@@ -28,6 +33,14 @@ class ScalarMetric(BaseMetric):
     value: Union[ScalarMetricValue, ScalarMetricConfidenceValue]
     aggregation: ScalarMetricAggregation = ScalarMetricAggregation.ARITHMETIC_MEAN
 
+    @validator('metric_name')
+    def validate_metric_name(cls, name: str):
+        clean_name = name.lower().strip()
+        if name.lower().strip() in RESERVED_METRIC_NAMES:
```
Comment (kozikkamil) [inline labelbox/data/annotation_types/metrics/scalar.py:39]: super nit, might want to change `name.lower().strip()` to `clean_name`

**Review by kozikkamil — APPROVED** [review]

**Review by msokoloff1 — APPROVED** [review]

**Pull request merged**